client/lib/media/store.js

```diff
@@ -123,6 +123,12 @@ MediaStore.dispatchToken = Dispatcher.register( function( payload ) {
 
 			MediaStore.emit( 'change' );
 			break;
+		case 'FETCH_MEDIA_LIMITS':
+			if ( ! action.siteId ) {
+				break;
+			}
+			MediaStore.emit( 'fetch-media-limits' );
```

Comment:
> You've committed a sin by emitting an event other than change here 

:smiling_imp:

> do we need a custom action type for FETCH_MEDIA_LIMITS

We don't but I wanted to avoid the duplicated lines in the switch cases. This should hopefully make it a little easier to trace this weird case as well.